Source/Core/Core/IOS/ES/NandUtils.cpp

```diff
@@ -49,12 +49,29 @@ ES::TMDReader ESDevice::FindInstalledTMD(u64 title_id, Ticks ticks) const
   return FindTMD(*m_ios.GetFSDevice(), Common::GetTMDFileName(title_id), ticks);
 }
 
-ES::TicketReader ESDevice::FindSignedTicket(u64 title_id) const
+ES::TicketReader ESDevice::FindSignedTicket(u64 title_id, std::optional<u8> desired_version) const
 {
-  const std::string path = Common::GetTicketFileName(title_id);
-  const auto ticket_file = m_ios.GetFS()->OpenFile(PID_KERNEL, PID_KERNEL, path, FS::Mode::Read);
+  std::string path = Common::GetTicketFileName(title_id);
+
+  // Check if desired version was passed
+  if (*desired_version == 1)
+    path = Common::GetV1TicketFileName(title_id);
+
```

Comment:
`std::string path = desired_version == 1 ? Common::GetV1TicketFileName(title_id) : Common::GetTicketFileName(title_id);`

Note that it should be `desired_version`, not `*desired_version`. Dereferencing an empty optional is undefined behaviour.